com.civicrm.hrjobroles/CRM/Hrjobroles/Upgrader.php

```diff
@@ -97,6 +97,91 @@ public function upgrade_1007() {
     return TRUE;
   }
 
+  /**
+   * Migrates existing job role funder to option values
+   *
+   * @return bool
+   */
+  public function upgrade_1008() {
+    $result = civicrm_api3('HrJobRoles', 'get', [
+      'return' => ['funder'],
+    ]);
+    $jobRoles = $result['values'];
+    $funderIds = $this->getFunderIds($jobRoles);
+    $contactFunders = $this->createFunderOptionValues($funderIds);
+    $this->updateJobRoleFunder($jobRoles, $contactFunders);
+
+    return TRUE;
+  }
+
+  /**
+   * Swaps job role funder contact id with option values and updates record
+   *
+   * @param array $jobRoles
+   * @param array $contactFunders
+   */
+  private function updateJobRoleFunder($jobRoles, $contactFunders) {
+    foreach ($jobRoles as $jobRole) {
+      foreach ($contactFunders as $key => $optionValue) {
+        $jobRole['funder'] = str_replace(
+          '|' . $key . '|',
+          '|' . $optionValue['value'] . '|',
+          $jobRole['funder']
+        );
+      }
+
+      civicrm_api3('HrJobRoles', 'create', [
+        'id' => $jobRole['id'],
+        'funder' => $jobRole['funder']
+      ]);
+    }
+  }
+
+  /**
+   * Sets up option values for job role funders
+   * using hrjc_funder option group
+   *
+   * @param array $funderIds
+   *
+   * @return array
+   */
+  private function createFunderOptionValues($funderIds) {
+    $funderOptionValueId = [];
+    $contactResult = civicrm_api3('Contact', 'get', [
+      'return' => ['sort_name', 'display_name'],
+      'id' => ['IN' => $funderIds],
+    ]);
+
+    $contacts = $contactResult['values'];
+    foreach ($contacts as $contact) {
+      $funderOptionValueId[$contact['id']] = CRM_Core_BAO_OptionValue::ensureOptionValueExists([
+        'option_group_id' => 'hrjc_funder',
+        'name' => $contact['display_name'],
+        'label' => $contact['sort_name'],
```

Comment:
Why was `display_name` chosen for `name` and `sort_name` for `label`?

Reply:
I played around the creation of contacts and modifying records from the db table. I discovered same values were inserted for both. After modifying the records from database, the sort_name was reflected in the label. Checking [job role](https://github.com/compucorp/civihr/blob/master/com.civicrm.hrjobroles/js/src/job-roles/services/job-role.service.js#L262), sort_name was also used for label. These formed my choice here. 🤷‍♂️ 
 Kindly advice if otherwise.

Reply:
Nope, that's fine. Just curious :-)